src/platform/web/Platform.js

```diff
@@ -143,7 +143,10 @@ export class Platform {
             this._serviceWorkerHandler.registerAndStart(assetPaths.serviceWorker);
         }
         this.notificationService = new NotificationService(this._serviceWorkerHandler, config.push);
-        this.crypto = new Crypto(cryptoExtras);
+        // `window.crypto.subtle` is only available in a secure context
+        if(window.isSecureContext) {
```

Comment:
I think I would rather see a solution to disable e2ee that makes it the explicit choice of the consuming app, rather than whether or not you happened to have activated https when deploying. The latter will be just as annoying to debug as it is now.
I think it's already sort of the idea that if you don't pass the olm paths in the platform options, that e2ee will be disabled, although this is very untested, so we can adopt that approach here. 

```suggestion
        if(this._assetPaths.olm) {
```